### PR TITLE
add editor support for pipelines, tasks and resources (#10)

### PR DESCRIPTION
**Pull request by lstocchi**
it is related to #10 

This PR allows to open a pipeline/task/pipeline resource in the editor and see its configuration

### CODE REVIEW

**Comment by lstocchi** [comment]
Changed a little bit how it works. By working on the updating/saving process i found out that working with LightVirtualFile doesn't trigger the onSave listener. Now a temporary file is created everytime a pipeline/task/resource is opened in the editor. The file is then deleted when the editor is closed. I also solved another problem bc before if you clicked on the same pipeline over and over again, it keeps opening several tabs with the same content. Now only one editor per component is opened.

File: src/main/java/com/redhat/devtools/intellij/common/actions/TknAction.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/common/actions/TknAction.java]: The common package is supposed to be extracted from this plugin and should contain only generic stuff. So every piece of code that is Tekton specific must be moved under the tektoncd sub packages

File: src/main/java/com/redhat/devtools/intellij/common/actions/component/OpenEditorTask.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import com.redhat.devtools.intellij.tektoncd.tkn.Tkn;
+import com.redhat.devtools.intellij.tektoncd.tree.PipelineNode;
+import com.redhat.devtools.intellij.tektoncd.tree.ResourceNode;
+import com.redhat.devtools.intellij.tektoncd.tree.TaskNode;
+
+import javax.swing.tree.TreePath;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class OpenEditorTask extends TknAction {
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/common/actions/component/OpenEditorTask.java]: Any reason why it does not followng the *Action name pattern

File: src/main/java/com/redhat/devtools/intellij/common/listener/FileEditorListener.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        if (tknProperty != null && tknProperty.equals("file")) {
+            ApplicationManager.getApplication().runWriteAction(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        if (Files.exists(Paths.get(file.getPath()))) {
+                            file.delete(this);
+                        }
+                    } catch (IOException e) {
+                        e.printStackTrace();
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/common/listener/FileEditorListener.java]: Replace by SLF4J API

**Comment by lstocchi** [comment]
Removed the listener and so the messageBus to deal with that. The deleteOnExit works fine. Great idea!! @jeffmaury :+1:

**Pull request merged**